File: dongyang_algo/day11/CompressAndFindShortestLength.py
# Python3 program to implement
# run length encoding
def printRLE(st):

    n = len(st)
    i = 0
    while i < n - 1:

        count = 1
        while (i < n - 1 and
                st[i] == st[i + 1]):
            count += 1
            i += 1
        i += 1

        print(str(count) + st[i - 1], end="")


if __name__ == "__main__":

    st = "aabbaccc"
    printRLE(st)

File: dongyang_algo/day11/CompressAndFindShortestLength.java
package day11;

public class CompressAndFindShortestLength {
    public static void main(String[] args) {
        String str = "aabbaccc";
        String result="";
        for (int i = 0; i < str.length(); i++) {
            int count = 1;
            while (i < str.length() - 1
                    && str.charAt(i) == str.charAt(i + 1)) {
                count++;
                i++;
            }
            result+=count+""+str.charAt(i);
        }
        System.out.println(result);
    }
}
